bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/GoveeHandler.java

```diff
@@ -318,8 +322,13 @@ public void updateDeviceState(@Nullable StatusResponse message) {
         if (newColorTempInKelvin != lastColorTempInKelvin) {
             logger.trace("Color-Temperature Status: old: {} K {}% vs new: {} K", lastColorTempInKelvin,
                     newColorTempInPercent, newColorTempInKelvin);
-            updateState(CHANNEL_COLOR_TEMPERATURE_ABS, new QuantityType<>(lastColorTempInKelvin, Units.KELVIN));
-            updateState(CHANNEL_COLOR_TEMPERATURE, new PercentType(newColorTempInPercent));
+            try {
+                updateState(CHANNEL_COLOR_TEMPERATURE_ABS, new QuantityType<>(lastColorTempInKelvin, Units.KELVIN));
+                updateState(CHANNEL_COLOR_TEMPERATURE, new PercentType(newColorTempInPercent));
+            } catch (IllegalArgumentException e) {
```

Comment:
Perhaps each state update should be wrapped separately into try/catch, so that failure to update color temperature in Kelvin doesn't prevent updating color temperature in percent?

I'm also wondering if the handler should verify the values itself, i.e. preventing the issue, rather than deliberately providing invalid values just to have `IllegalArgumentException` thrown. The binding defines the valid range as 2000-9000:
https://github.com/openhab/openhab-addons/blob/35add30e0c9e69d4d246725d207ee45ced80c59b/bundles/org.openhab.binding.govee/src/main/resources/OH-INF/thing/thing-types.xml#L28

so this could be easily validated.

Also, if the received color temperature in Kelvin is something like 9096 (from the issue), perhaps the defined range here is wrong?

Reply:
Changed it so that each update has its own try/catch block.

Previously i considered to pre-check values, but decided not to. Now iyou mention it, i changed my mind and removed the try block for local validation.